food-truck-api/src/main/java/food/truck/api/endpoint/TruckEndpoint.java

```diff
@@ -226,4 +233,58 @@ public void deleteTruckRouteLocations(@AuthenticationPrincipal User user, @Reque
     public Route getTodaysRoute(@PathVariable long truckId) {
         return truckService.getActiveRoute(truckId, LocalDateTime.now().getDayOfWeek());
     }
+
+    @Secured({"ROLE_USER"})
+    @PostMapping("/truck/{truckId}/subscribe")
+    public Optional<SubscriptionView> subscribe(@AuthenticationPrincipal User u, @PathVariable long truckId) {
+        var t = truckService.findTruckById(truckId);
+        if (t.isEmpty()) {
+            return Optional.empty();
+        }
+        var truck = t.get();
+
+        // Prevent an owner from subscribing to their own truck.
+        if (!u.getId().equals(truck.getUserId())) {
+            var sub = new Subscription();
+            sub.setTruck(truck);
+            sub.setUser(u);
+
+            return Optional.of(subscriptionService.saveSubscription(sub)).map(SubscriptionView::of);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    @Secured({"ROLE_USER"})
+    @GetMapping("/truck/{truckId}/subscription")
+    public Optional<SubscriptionView> subscribedToTruck(@AuthenticationPrincipal User u, @PathVariable long truckId) {
+        var subs = subscriptionService.findSubsByUser(u);
+        var iter = subs.stream()
+            .filter(sub -> sub.getTruck().getId().equals(truckId))
+            .map(SubscriptionView::of)
+            .iterator();
+        if (iter.hasNext()) {
+            return Optional.of(iter.next());
+        }
+        return Optional.empty();
+    }
+
+    @Secured({"ROLE_USER"})
+    @DeleteMapping("/truck/{truckId}/unsubscribe")
+    public void unsubscribe(@AuthenticationPrincipal User u, @PathVariable long truckId) {
+        var t = truckService.findTruckById(truckId);
+        if (t.isEmpty()) {
+            return;
```

Comment:
Not sure if 100% swallowing the error is the best approach.

Reply:
According to the idempotent spec, this should return a 404 starting with the second time you make this request in a row, but I wasn't sure how to do that here.

Reply:
Returning a 404? You can just throw a `new ResponseStatusException(HttpStatus.NOT_FOUND, "message")`, and that'll auto-return a 404 to the caller.

If you're referring to checking when they've viewed it twice, that'd require storing a flag in the database, and would feel almost like a hack given we're writing a RESTful service. But I don't know.